packages/adblocker/src/engine/engine.ts

```diff
@@ -79,6 +79,12 @@ export interface BlockingResponse {
         contentType: string;
         dataUrl: string;
       };
+  substitude:
+    | undefined
+    | {
+        modifiedUrl: string | undefined;
+        filters: Map<NetworkFilter, NetworkFilter | undefined>;
```

Comment:
Why return a `Map` here?

Reply:
It implies exceptions. I should have more verbosity in the code here.